### PR TITLE
Use Plugin::load() when creating classless plugins.

### PR DESCRIPTION
**Pull request by markstory**
I completely missed that Plugin::load() does a bunch of work to generate the correct paths for plugins. By using Plugin::load() and fetching the created plugin we can leverage the same logic. There is an argument to be had around deprecating Plugin::load(). Given the troubles plugin classes have given us I'm not comfortable in doing that right now.

Refs #12271

### CODE REVIEW

**Comment by saeideng** [comment]
can we duplicate code ?
 `Plugin::load()` has problem with calling bootstrap directly

**Comment by markstory** [comment]
Darn I forgot about that. Duplicating code will allow us to deprecate plugin::load() more easily in the future as well I guess.

**Comment by markstory** [comment]
@saeideng I think I've resolved the issue you raised and also fixed a potential problem where `cakephp-plugins.php` would not get loaded.

File: src/Http/BaseApplication.php [hunk 1]
```diff
@@ -119,22 +123,22 @@ public function getPlugins()
      * @param array $config Configuration options for the plugin
      * @return \Cake\Core\PluginInterface
      */
-    public function makePlugin($name, array $config)
+    protected function makePlugin($name, array $config)
```
Comment (markstory) [inline src/Http/BaseApplication.php]: There is no reason this method needs to be public, and I consider it a mistake that I would like to fix now vs. later. If folks think it is a risky change I'm ok with reverting this.

Comment (ADmad) [reply src/Http/BaseApplication.php]: Making is protected sounds good to me.

**Pull request merged**